### PR TITLE
JIT: Remove old fgRemoveRefPred

### PR DESCRIPTION
**Pull request by amanasifkhalid**
Part of #93020. Removes the old `fgRemoveRefPred` implementation in favor of the new one that avoids calling `fgGetPredForBlock` unless the successor block's pred list needs to be updated.

### CODE REVIEW

**Comment by amanasifkhalid** [comment]
cc @dotnet/jit-contrib, @AndyAyersMS PTAL. [Small diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=583930&view=ms.vss-build-web.run-extensions-tab) from changes in profile data: Previously, `fgRemoveRefPred` would return the removed edge only if its dup count is now zero. Now, since we already have access to the edge being removed, we can always pass the removed edge to `fgAddRefPred`, transferring the old edge's likelihood/weight over.

File: src/coreclr/jit/fgopt.cpp [hunk 1]
```diff
@@ -1566,25 +1566,31 @@ bool Compiler::fgOptimizeBranchToEmptyUnconditional(BasicBlock* block, BasicBloc
...
         // Optimize the JUMP to empty unconditional JUMP to go to the new target
-        FlowEdge* const newEdge = fgAddRefPred(bDest->GetTarget(), block, fgRemoveRefPred(bDest, block));
-
         switch (block->GetKind())
         {
             case BBJ_ALWAYS:
             case BBJ_CALLFINALLYRET:
+            {
+                fgRemoveRefPred(block->GetTargetEdge());
+                FlowEdge* const newEdge = fgAddRefPred(bDest->GetTarget(), block, block->GetTargetEdge());
```
Comment (AndyAyersMS) [inline src/coreclr/jit/fgopt.cpp:1575]: I kind of prefer

```suggestion
                FlowEdge* const oldEdge = block->GetTargetEdge();
                fgRemoveRefPred(oldEdge);
                FlowEdge* const newEdge = fgAddRefPred(bDest->GetTarget(), block, oldEdge);
```
In the long run all this probably turns into a call to a changeXXtarget method.

Comment (AndyAyersMS) [reply src/coreclr/jit/fgopt.cpp:1575]: Ditto for the two cases below

**Review by AndyAyersMS — APPROVED** [review]

File: src/coreclr/jit/fgopt.cpp (hunk 1 above)
Comment (amanasifkhalid) [reply src/coreclr/jit/fgopt.cpp:1575]: I agree that looks better.

> In the long run all this probably turns into a call to a changeXXtarget method.

That's my plan -- I'll make sure those new methods use your proposed format.

**Pull request merged**